qiime_studio/api/v1.py

```diff
@@ -149,35 +137,44 @@ def execute_workflow(plugin_name, workflow_name):
             elif type_ == 'param':
                 parameters[name] = value
             elif type_ == 'out':
-                if not value.endswith('.qzf'):
-                    value += '.qzf'
+                if not value.endswith('.qza'):
+                    value += '.qza'
                 outputs[name] = os.path.join(path, value)
 
-    def toggle_completion(future_result):
-        future_id = id(future_result)
+    def toggle_completion(future_result, job):
+        job_id = str(job.uuid)
         completed_future = future_result.result()
-        if completed_future.returncode != 0:
-            __JOBS[future_id]['error'] = True
-            __JOBS[future_id]['message'] = \
-                completed_future.stderr.decode('utf-8')
-        __JOBS[future_id]['completed'] = True
+        __JOBS[job_id]['error'] = completed_future.returncode is not 0
+        __JOBS[job_id]['stderr'] = completed_future.stderr.decode('utf-8')
+        __JOBS[job_id]['stdout'] = completed_future.stdout.decode('utf-8')
+        __JOBS[job_id]['completed'] = True
+        __JOBS[job_id]['finished'] = '{:%Y-%b-%d %H:%M:%S}' \
+                                     .format(datetime.datetime.now())
 
     now = '{:%Y-%b-%d %H:%M:%S}'.format(datetime.datetime.now())
-    future_result = SUBPROCESS_EXECUTOR(workflow, inputs, parameters, outputs)
-    future_id = id(future_result)
-    __JOBS[future_id] = {
+    future_result, job = SUBPROCESS_EXECUTOR(workflow,
+                                             inputs,
+                                             parameters,
+                                             outputs)
+    __JOBS[str(job.uuid)] = {
         'completed': False,
         'error': False,
-        'message': ''
+        'finished': None,
+        'started': now
     }
-    future_result.add_done_callback(toggle_completion)
+    future_result.add_done_callback(lambda future_result:
+                                    toggle_completion(future_result, job))
```

Comment:
Thanks! Now that we were using the job info to store it in the dictionary it needed a way to get both pieces in the callback.